### PR TITLE
Deprecations for 2.0

### LINKED ISSUE #8964: remove 2.0 deprecations
Everything except tile provider stuff, which will be in a separate PR.

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #8964
- [x] release document entry (if new feature or API change)

This does not quite removed everything: 

* tile providers will go on their own PR
* remove `.callback` throughout will take some work, so separate PR there too

@mattpap I would appreciate a review of this since it attempts to remove all the coffee script machinery, also I want to confirm that the `Dropdown` properties really should be removed.

### CODE REVIEW

File: bokeh/util/compiler.py [hunk 1]
```diff
@@ -278,8 +241,6 @@ def implementation(self):
         if isinstance(impl, str):
             if "\n" not in impl and impl.endswith(exts):
                 impl = FromFile(impl if isabs(impl) else join(self.path, impl))
-            else:
-                impl = CoffeeScript(impl)
```
Comment (bryevdv) [inline bokeh/util/compiler.py]: Should this default to `TypeScript` now?

**Comment by bryevdv** [comment]
@mattpap please take a look tomorrow. I would like to merge tomorrow night (keeping it rebased on `landing-2.0` is a slight chore)

File: bokeh/util/compiler.py (hunk 1 above)
Comment (mattpap) [reply bokeh/util/compiler.py]: Yes.

File: bokehjs/src/compiler/compile.ts [hunk 2]
```diff
@@ -240,7 +203,7 @@ const compile_and_resolve_deps = (input: {code: string, lang: string, file: stri
 if (argv.file != null) {
   const input = {
     code: fs.readFileSync(argv.file as string, "utf-8"),
-    lang: (argv.lang as string | undefined) || "coffeescript",
+    lang: argv.lang as string,
```
Comment (mattpap) [inline bokehjs/src/compiler/compile.ts]: The original needs to be restored with `coffeescript` replaced with `typescript`.

**Pull request merged**